src/main/java/uk/gov/hmcts/darts/automation/pageObjects/Search.java

```diff
@@ -0,0 +1,243 @@
+package uk.gov.hmcts.darts.automation.pageObjects;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class Search {
+
+    private static WebDriver webDriver;
+
+    public Search(WebDriver driver) {
+        Search.webDriver = driver;
+
+    }
+
+    public enum SearchResultsEnum {
+        CASEID,
+        COURTHOUSE,
+        COURTROOM,
+        JUDGES,
+        DEFENDANTS,
+        INVALID
+    }
+
+    public Table<Integer, Integer, String> scrapeTableData(WebElement table)
+    {
+        Table<Integer, Integer, String> tableObject = HashBasedTable.create();
+
+        List<WebElement> rows = table.findElements(By.tagName("tr"));
+
+        int rowIndex = 0;
+        int colIndex = 0;
+        for (WebElement row : rows) {
+            List<WebElement> cols = row.findElements(By.tagName("td"));
+            for (WebElement col : cols) {
+                tableObject.put(rowIndex, colIndex++, col.getText());
+            }
+            rowIndex++;
+            colIndex = 0;
+        }
+
+        return tableObject;
+    }
+
+    //used to validate table search result
+    public void validateExactSearchResults(List<Map<String, String>> expectedSearchResults) {
+
+        WebElement searchResultsTable = getSearchResultsTable();
+        //get table rows
+        Table<Integer, Integer, String> searchTable = scrapeTableData(searchResultsTable);
+
+        List<List<String>> expectedSearchRowList = new ArrayList<>();
+        expectedSearchResults.forEach(
+                expectedTableRow -> {
+                    List<String> cellsList = new ArrayList<>();
+                    String caseId = expectedTableRow.get("CaseID");
+                    cellsList.add(caseId);
+                    String courthouse = expectedTableRow.get("Courthouse");
+                    cellsList.add(courthouse);
+                    String courtRoom = expectedTableRow.get("Courtroom");
+                    cellsList.add(courtRoom);
+                    String judges = expectedTableRow.get("Judges");
+                    cellsList.add(judges);
+                    String defendants = expectedTableRow.get("Defendants");
+                    cellsList.add(defendants);
+                    //add row
+                    expectedSearchRowList.add(cellsList);
+                });
+
+        //start with rowCounter = 0 as expected table doesn't have header row
+        for (int rowCounter = 0; rowCounter < expectedSearchRowList.size(); rowCounter++) {
+            int actualRowCounter = rowCounter+1;//skip header row
+            String actualCaseId = searchTable.get(actualRowCounter, SearchResultsEnum.CASEID.ordinal());
+            //"Case Id" is first, so use index "0"
```

Comment:
no need to add comments all over the place. You ok to remove please?

Reply:
Yes, sure, it was more for guidance over the course of creation and looking out for commented code, sorting now, thanks.

Reply:
That is now done and pushed.